src/main/java/org/example/dao/PointDAOImpl.java

```diff
@@ -0,0 +1,95 @@
+package org.example.dao;
+
+import org.example.configurations.ConnectionPool;
+import org.example.model.Point;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class PointDAOImpl implements PointDAO {
+    private static final ConnectionPool CONNECTIONPOOL = ConnectionPool.getInstance();
+
+    public PointDAOImpl() {
+    }
+
+    @Override
+    public void insertPoint(Point point) {
+        Connection connection = CONNECTIONPOOL.getConnectionFromPool();
+        try (PreparedStatement statement = connection.prepareStatement("INSERT INTO points (id, xCoordinate, yCoordinate) VALUES (?, ?, ?)", Statement.RETURN_GENERATED_KEYS)) {
```

Comment:
We don't need to inset an ID here since we are using generated keys here so the query should be
INSERT INTO points (x_coordinate, y_coordinate) VALUES (?, ?)